### PR TITLE
Port gps support to QSerialPort (byebye QExtSerialPort)

### PR DESCRIPTION
**Pull request by jef-n**
removes the internal copy of QExtSerialPort.  Not sure if we even need QSerialPort anymore.  Qt's [QtPositioning](https://doc.qt.io/qt-5.10/qtpositioning-index.html) apparently can work with NMEA speaking serial receivers (at least on Windows), internal GPS on Windows 10.

Untested otherwise - receiver is in the office...  So probably not good to merge.

### CODE REVIEW

**Comment by m-kuhn** [comment]
Saying goodbye is never easy. But in this case here I wouldn't have any troubles.
@jef-n time to let go?

**Comment by nyalldawson** [comment]
What's the go here? Had a chance to test?

**Comment by nyalldawson** [comment]
Ping @probot

File: src/providers/postgres/qgspostgresprovider.cpp [hunk 1]
```diff
@@ -2158,7 +2158,7 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
       for ( int i = 0; i < fieldId.size(); i++ )
       {
         int attrIdx = fieldId[i];
-        QVariant value = attrIdx < attrs.length() ? attrs.at( attrIdx ) : QVariant( QVariant::Int );
+        QVariant value = attrs.value( attrIdx, QVariant( QVariant::Int ) );
```
Comment (3nids) [inline src/providers/postgres/qgspostgresprovider.cpp]: I guess this is not related...?

File: src/core/qgscoordinatereferencesystem.cpp [hunk 2]
```diff
@@ -2111,7 +2111,11 @@ int QgsCoordinateReferenceSystem::syncDatabase()
     return -1;
   }
 
+#ifdef QGISDEBUG
```
Comment (3nids) [inline src/core/qgscoordinatereferencesystem.cpp]: neither this one

**Comment by 3nids** [comment]
Even if not properly tested, I'd be in favor or merging this to remove old libs dependency.

File: .docker/qgis3-build-deps.dockerfile [hunk 3]
```diff
@@ -45,6 +45,7 @@ RUN  apt-get update \
     libqt5webkit5-dev \
     libqt5xml5 \
     libqt5xmlpatterns5-dev \
+    libqt5serialport5-dev \
```
[the remark this answers is not in the document]
Comment (jef-n) [reply .docker/qgis3-build-deps.dockerfile]: @Gustry yes and no.  INSTALL is updated by target `t2tdoc`, the package list there comes from `doc/linux.t2t`, which is updated by running `scripts/scandeps.pl` from the debian packaging information in `debian/`.
Done in 8c751f8b0bb.

[the remark this answers is not in the document]
Comment (Gustry) [reply .docker/qgis3-build-deps.dockerfile]: Yes I knew that it was coming from another file ;-)
That's why I didn't fix my self by making a PR. I was letting you doing all these intermediate steps to have it in `INSTALL` ;-)

Thanks for the fix !
